pkg/dns/etcd_dns.go

```diff
@@ -118,10 +130,8 @@ func (c *coreDNS) list(key string) ([]SrvRecord, error) {
 		// In all other situations when we want to list all DNS records,
 		// which is handled in the else clause.
 		for _, domainName := range c.domainNames {
-			if key != msg.Path(fmt.Sprintf(".%s.", domainName), defaultPrefixPath) {
-				if srvRecord.Key == "/" {
-					srvRecords = append(srvRecords, srvRecord)
-				}
+			if key != "" && key != msg.Path(fmt.Sprintf(".%s.", domainName), defaultPrefixPath) {
+				srvRecords = append(srvRecords, srvRecord)
```

Comment:
why do you need this `if` check here? `srvRecords = append(srvRecords, srvRecord)` is executed in all scenarios.

Reply:
We need to only return back srvRecords which match the key otherwise we will have unexpected entries.